frontend/public/components/utils/tile-view-page.jsx

```diff
@@ -0,0 +1,686 @@
+  const newFilterCounts = {};
+
+  _.each(filterGroups, filterGroup => {
+    const count = _.countBy(activeItems, filterGroup);
+    _.each(Object.keys(count), valueKey => {
+      _.set(newFilterCounts, [filterGroup, valueKey], (count[valueKey] || 0));
+    });
+  });
+
+  return newFilterCounts;
+};
+
+const setURLParams = params => {
+  const location = window.location;
+  const url = new URL(location);
+  const searchParams = `?${params.toString()}${url.hash}`;
+
+  history.replace(`${url.pathname}${searchParams}`);
+};
+
+const updateURLParams = (filterName, value) => {
+  const params = new URLSearchParams(window.location.search);
+
+  if (value) {
+    params.set(filterName, Array.isArray(value) ? JSON.stringify(value) : value);
+  } else {
+    params.delete(filterName);
+  }
+  setURLParams(params);
+};
+
+const clearFilterURLParams = selectedCategoryId => {
+  const params = new URLSearchParams();
+
+  if (selectedCategoryId) {
+    params.set(CATEGORY_URL_PARAM, selectedCategoryId);
+  }
+
+  setURLParams(params);
+};
+
+const getActiveValuesFromURL = (availableFilters, filterGroups) => {
+  const searchParams = new URLSearchParams(window.location.search);
+  const categoryParam = searchParams.get(CATEGORY_URL_PARAM);
+  const keywordFilter = searchParams.get(KEYWORD_URL_PARAM);
+
+  const selectedCategoryId = categoryParam || 'all';
+
+  const groupFilters = {};
+
+  _.each(filterGroups, filterGroup => {
+    const groupFilterParam = searchParams.get(filterGroup);
+    if (!groupFilterParam) {
+      return;
+    }
+
+    try {
+      _.set(groupFilters, [filterGroup], JSON.parse(groupFilterParam));
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.warn('could not update filters from url params: could not parse search params', e);
+    }
+  });
+
+  const activeFilters = getActiveFilters(keywordFilter, groupFilters, availableFilters);
+
+  return {selectedCategoryId, activeFilters};
+};
+
+const getFilterSearchParam = groupFilter => {
+  const activeValues = _.reduce(_.keys(groupFilter), (result, typeKey) => {
+    return groupFilter[typeKey].active ? result.concat(typeKey) : result;
+  }, []);
+
+  return _.isEmpty(activeValues) ? '' : JSON.stringify(activeValues);
+};
+
+const defaultFilters = {
+  keyword: {
+    value: '',
+    active: false,
+  },
+};
+
+export class TileViewPage extends React.Component {
+  constructor(props) {
+    super(props);
+    const {items, itemsSorter, filterGroups, getAvailableCategories, getAvailableFilters} = this.props;
+
+    // Filters are populated based on white list of item properties
+    const availableFilters = getAvailableFilters(defaultFilters, items, filterGroups);
+
+    const categories = getAvailableCategories(items);
+
+    this.state = {
+      categories: categorizeItems(items, itemsSorter, categories),
+      selectedCategoryId: 'all',
+      availableFilters,
+      activeFilters: availableFilters,
```

Comment:
I'm a bit confused about the distinction between `activeFilters` and `availableFilters`, could you explain?

Reply:
Available filters are the set of filters that can be used. ActiveFilters are the current state of the filters.